py/dml/c_backend.py

```diff
@@ -354,6 +354,8 @@ def generate_hfile(device, headers, filename):
         + '('+crep.structtype(device)+' *_obj);\n')
     out('void soft_reset_'+crep.cname(device)
         + '('+crep.structtype(device)+' *_obj);\n')
+    out('int _dml_gdb_marker_'
```

Comment:
this declaration is still redundant, since you also have `generate_marker`? (Also, it would not surprise me if this causes `--split-c-file` to break)

Reply:
Huh! Y'know, I was about to contradict you since I tested this earlier, but now I tested it again and sure enough, the declaration in the .h file is unnecessary. Weird.

About `--split-c-file`; yeah, I have no idea how this interacts with it, but I also wouldn't be surprised if it breaks. I'll test. If it doesn't work, how do we solve it? Perhaps we shouldn't? Add a documentation string and check stating that `--split-c-file` is incompatible with `-g`?

Reply:
It will or won't work, depending on how you use `--split-c-file`. The easy way is to have a single-class module and just link all .c files together; this should be unproblematic as long as you don't touch the .h file. This is what t126 tests. The harder way is to insert an extra link step that combines all .o files of one device into one big .o, while hiding all symbols except the `initialize_<classname>` function, to allow multiple DML classes in one module. The GDB symbol will vanish here unless you update the exportmap.

Speaking of which -- perhaps we should reuse the `initialize_*` function as dml-gdb marker instead of adding a new one?

Reply:
> Speaking of which -- perhaps we should reuse the initialize_* function as dml-gdb marker instead of adding a new one?

That was the first thing I tried, didn't get it to work -- I figured either because the function's static and/or because it isn't declared in the .h file. Are you suggesting we change that?